src/main.cpp

```diff
@@ -473,6 +477,31 @@ static void _assert_permissions(const string &sota_config_dir)
 	unlink(test_file.c_str());
 }
 
+static void _assert_not_registered(const string &sota_config_dir)
+{
+	string path = sota_config_dir + "/sql.db";
+	if (access(path.c_str(), F_OK ) == 0 ) {
+		cerr << "ERROR: Device appears to already be registered in " << path << endl;
+		exit(EXIT_FAILURE);
+	}
+}
+
+static void _assert_not_running() {
+	string out = _spawn("ps -ef");
```

Comment:
I wonder if we could leverage the mechanism that aktualizr uses, it sets lock (flock) on storage.lock to check if there is another aktualizr instance running. Unfortunately, lmp-device-register doesn't share code base with aklite, otherwise, we could make use of `StorageLock` class.

Reply:
If there is a generic and ak-lite thing to check (such as storage.lock), I would prefer that instead as the ps behavior can change depending on the rootfs (busybox x procps).

Reply:
The problem I was finding was that assume you have to look in /var/sota. This way is ugly, but it finds *any* copy of the daemon running, no matter how its been configured.

Reply:
>The problem I was finding was that assume you have to look in /var/sota

`"/var/sota"` is hardcoded into the lmp-device-register code anyway. Nevertheless, I think, it's good enough to be merged, because all my ideal solutions arrive at merging lmp-device-register code to aklite code :)

Reply:
Its defaulted, but not hard-coded - https://github.com/foundriesio/lmp-device-register/blob/72b145e4de2579105ee873d98f4c6962dad8067a/src/main.cpp#L83